algo/src/test/kotlin/com/hartwig/actin/algo/evaluation/treatment/SystemicTreatmentAnalyserTest.kt

```diff
@@ -92,34 +93,59 @@ class SystemicTreatmentAnalyserTest {
-        assertNameForLastSystemicTreatmentHistoryEntry(treatmentHistory, "1")
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "1", ::lastSystemicTreatment)
 
         treatmentHistory.add(treatmentHistoryEntry(setOf(treatment("2", true)), 2021))
-        assertNameForLastSystemicTreatmentHistoryEntry(treatmentHistory, "2")
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "2", ::lastSystemicTreatment)
 
         treatmentHistory.add(treatmentHistoryEntry(setOf(treatment("3", true)), 2021, 1))
-        assertNameForLastSystemicTreatmentHistoryEntry(treatmentHistory, "3")
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "3", ::lastSystemicTreatment)
 
         treatmentHistory.add(treatmentHistoryEntry(setOf(treatment("4", true)), 2021, 10))
-        assertNameForLastSystemicTreatmentHistoryEntry(treatmentHistory, "4")
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "4", ::lastSystemicTreatment)
 
         treatmentHistory.add(treatmentHistoryEntry(setOf(treatment("5", true)), 2021, 8))
-        assertNameForLastSystemicTreatmentHistoryEntry(treatmentHistory, "4")
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "4", ::lastSystemicTreatment)
     }
 
-    private fun assertNameForLastSystemicTreatmentHistoryEntry(treatmentHistory: List<TreatmentHistoryEntry>, name: String) {
-        assertThat(lastSystemicTreatment(treatmentHistory)!!.treatments.first().name).isEqualTo(name)
+    @Test
+    fun `Should determine first systemic treatment`() {
+        val treatmentHistory = mutableListOf(treatmentHistoryEntry(setOf(treatment("1", true)), 2020, 5))
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "1", ::firstSystemicTreatment)
+
+        treatmentHistory.add(treatmentHistoryEntry(setOf(treatment("2", true)), 2020))
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "2", ::firstSystemicTreatment)
+
+        treatmentHistory.add(treatmentHistoryEntry(setOf(treatment("3", true)), 2019, 12))
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "3", ::firstSystemicTreatment)
+
+        treatmentHistory.add(treatmentHistoryEntry(setOf(treatment("4", true)), 2019, 11))
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "4", ::firstSystemicTreatment)
+
+        treatmentHistory.add(treatmentHistoryEntry(setOf(treatment("5", true)), 2020, 1))
+        assertNameForSystemicTreatmentHistoryEntry(treatmentHistory, "4", ::firstSystemicTreatment)
+    }
+
+    private fun assertNameForSystemicTreatmentHistoryEntry(
+        treatmentHistory: List<TreatmentHistoryEntry>,
+        name: String,
+        treatmentSelector: (List<TreatmentHistoryEntry>) -> TreatmentHistoryEntry?
+    ) {
+        val treatment = treatmentSelector(treatmentHistory)
+        assertThat(treatment!!.treatments.first().name).isEqualTo(name)
```

Comment:
You could inline here and keep it null-safe:
```suggestion
        assertThat(treatmentSelector(treatmentHistory)?.treatments?.first()?.name).isEqualTo(name)
```